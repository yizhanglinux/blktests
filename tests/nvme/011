#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# This is a data verification test for file backed ns.

. tests/nvme/rc

DESCRIPTION="run data verification fio job on NVMeOF file-backed ns"
TIMED=1

requires() {
	_nvme_requires
	_have_fio
	_require_nvme_trtype_is_fabrics
}

set_conditions() {
	_set_nvme_trtype "$@"
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local ns

	_nvmet_target_setup --blkdev file

	_nvme_connect_subsys

	ns=$(_find_nvme_ns "${def_subsys_uuid}")

	_run_fio_verify_io --size="${nvme_img_size}" \
		--filename="/dev/${ns}"

	_nvme_disconnect_subsys

	_nvmet_target_cleanup

	echo "Test complete"
}
