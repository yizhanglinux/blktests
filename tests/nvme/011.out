Running nvme/011
disconnected 1 controller(s)
Test complete
